### ISSUE TITLE
Additional Parameters "Scheduler args" NOT Saved in Toml file AND NOT passed

### ISSUE THREAD

**Issue by rafstahelin** [bmaltais/kohya_ss#2527]
@bmaltais would it be possible to add "additional parameters" to toml structure? There's no line for this important parameter

**Comment by rafstahelin** [comment]
ok, i did some digging. Seems like the problem is talked about here:
[https://github.com/kohya-ss/sd-scripts/issues/601#issuecomment-1605241097](url)

Seems like additional parameters needs to be passed as str rather than list in toml

I am now running these additional args in cli since they are ignored in toml

@bmaltais is there any way to adjust this?

**Comment by bmaltais** [comment]
Can you provide an example of that specific toml variable as what it is setting it as now... and what it should be set to?

I should be able to transform to the right format based on the example...

**Comment by rafstahelin** [comment]
@bmaltais 
let me rephrase. I'd like to pass:
`--lr_scheduler_type "CosineAnnealingLR" --lr_scheduler_args "T_max=250"` in GUI

But at the moment, it only works in CLI. If you place it in GUI Additional Parameters it wont work

See:
```
@DarkAlchy Just pass --lr_scheduler_type and --lr_scheduler_args to command line.

Most of schedulers in torch.optim.lr_scheduler can be imported directly.

For example:
accelerate launch "train_network.py" ... --lr_scheduler_type "CosineAnnealingLR" --lr_scheduler_args "T_max=100"
```
Here's the thread: [https://github.com/kohya-ss/sd-scripts/issues/601#issuecomment-1602496423](url)

Seems like these args need to be passed as string like in CLI, but toml is list. 

--------

I don't know if this is the answer, but not sure it was ever addressed:
[https://github.com/kohya-ss/sd-scripts/issues/601#issuecomment-1605241097](url)
```
@DarkAlchy Oh, my fault.

It seems that args.blahblah just passed str instead of a list through .toml.

We should pass a list through .toml:
lr_scheduler_args = ['T_max=100', ...]
```
Could you somehow run the arg in CLI (necessary) but place it in toml in a way that it passes to wandb at the same time?

Does this make sense. 

2 results:
1. additional parameter scheduling works through GUI
2. . the parameter passes to wandb

**Comment by bmaltais** [comment]
Ok, I can work with that to reproduce the issue and find a fix… hope it won’t lead to a rat’s nest if code change.

**Comment by rafstahelin** [comment]
> Ok, I can work with that to reproduce the issue and find a fix… hope it won’t lead to a rat’s nest if code change.

I know everyone using scheduling args through GUI and that want to pass to wandb will be appreciative

**Comment by bmaltais** [comment]
OK, It is now available in the `dev` branch:

![image](https://github.com/bmaltais/kohya_ss/assets/7474674/13a50edc-2adc-4bad-9555-c3fdde0fc7d3)

I have added it to the `./test/config/dreambooth-AdamW8bit-toml.json` test config so you can easilly see it in action.

**Comment by rafstahelin** [comment]
@bmaltais confirming, this is working perfectly. Thanks!

![image](https://github.com/bmaltais/kohya_ss/assets/64715158/fad762cc-cf17-4ff8-a0fb-f4b61fd8f6ad)
